### PR TITLE
news: Hexo 4.0.0 released

### PR DESCRIPTION
**Pull request by curbengh**
- [x] Others (Update, fix, translation, etc...)

### CODE REVIEW

**Comment by curbengh** [comment]
Pending https://github.com/hexojs/hexo/pull/3695

**Comment by SukkaW** [comment]
We might write something about what we have optimized, maybe in a separate post.
This is what tommy351 written for Hexo 3.2 release.
https://hexo.io/news/2016/02/28/hexo-3-2-released/

**Comment by curbengh** [comment]
Good idea, I concur with writing it in a separate post.

**Comment by tomap** [comment]
And maybe add a breaking change section, like node 6, possible changes in url path, possible changes in url encode, ...?

**Comment by curbengh** [comment]
The section is [already there](https://github.com/hexojs/site/blob/a9beb2ce68a6c18eee98cb0fa694a97460664775/source/_posts/2019-10-10-hexo-4-released.md#breaking-change).

File: source/_posts/2019-10-14-hexo-4-released.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  * use this helper `full_url_for(page.path)` instead of `config.url + page.path`
+- fix: external_link should use after_render [#3675]
+  * added `field` and `exclude` options, see [#3675] for guide.
+
+## Fix
+
+- fix: set english as default [#3654]
+- perf(meta_generator): drop cheerio [#3671]
+- perf(open_graph): drop cheerio and use regex [#3680]
+- perf(external_link): drop cheerio and use regex [#3685]
```
Comment (YoshinoriN) [inline source/_posts/2019-10-14-hexo-4-released.md]: IMHO these changes divide into another section is better. (e.g `Improvement` or `Performance` etc...).

File: source/_posts/2019-10-14-hexo-4-released.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+## Breaking change
+
+- chore: drop Node 6 [#3598]
+- fix post_link, asset_link when title contains unescaped html charaters [#3704]
+  * Affects `asset_link`, `post_link` [tag plugins](https://hexo.io/docs/tag-plugins]
+  * If you want to retain unescaped characters, set `false` to the final argument `{% asset_link 'filename 'title' 'false' %}
+- fix: encode permalink by default [#3708]
+  * If you currently use `encodeURI(post.permalink)` (including `permalink` of page, tag & category variables), there are three options:
+    1. Use `encodeURI(decodeURI(post.permalink))` for backward-compatibility with hexo v3.9 (and older]
```
Comment (YoshinoriN) [inline source/_posts/2019-10-14-hexo-4-released.md]:
```diff
- (and older]
+ (and older)
```

File: source/_posts/2019-10-14-hexo-4-released.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    - [`decodeURL()`](https://github.com/hexojs/hexo-util#decodeurlstr) of hexo-util can decode [punycoded](https://en.wikipedia.org/wiki/Punycode) domain.
+  * This change does not apply to `this.url` variable.
+- fix(paginator): add `escape` option for compatibility with hexo-util@1.3.0 [#3728]
+  * Theme devs, if you customize `prev_text`/`next_text` with html (e.g. to insert icons), you need to set [`escape: false`](https://github.com/theme-next/hexo-theme-next/blob/7d6272274afcc88edcefe3504e452a502c05329f/layout/_partials/pagination.swig#L8) in the parameter.
+
+## Feature
+
+- feat(filter): use existing excerpt if possible [#3612]
+  * Support `excerpt:` in [front-matter](https://hexo.io/docs/front-matter)
+- switch minimatch to micromatch [#3538]
```
Comment (YoshinoriN) [inline source/_posts/2019-10-14-hexo-4-released.md]: Missing link to `[#3538]`

File: source/_posts/2019-10-14-hexo-4-released.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  * Refer to [micromatch docs](https://github.com/micromatch/micromatch#extended-globbing) for advanced globbing
+- feat: add option to disable meta generator tag [#3653]
+  * Add `meta_generator: false` to config to disable
+- feat(generator): allow limit parallel generation [#3665]
```
Comment (YoshinoriN) [inline source/_posts/2019-10-14-hexo-4-released.md]: Missing link to `[#3665]`

File: source/_posts/2019-10-14-hexo-4-released.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+- perf(external_link): drop cheerio and use regex [#3685]
+- Use filename when title is not specified in the front-matter [#3672]
+- fix: ignore categories / tags with zero posts [#3624]
+- fix(open_graph): remove index.html from url [#3661]
+- fix(helpers, tag plugins): encode url by default [#3710]
+  * Refer to [#3708] for guide on decoding.
+- fix(open_graph): percent-encode url, not html escape [#3686]
+- perf(cache): Improve cache, Reduce Memory Usages [#3756]
+- fix: Allow backtick code block in "blockquote" tag plugin [#2321]
+- fix: Correct processing of backtick code block on blockquote [#3765]
```
Comment (YoshinoriN) [inline source/_posts/2019-10-14-hexo-4-released.md]: Missing link to `[#3765]`

**Pull request merged**